runelite-client/src/test/java/net/runelite/client/plugins/kourendlibrary/KourendLibraryTest.java

```diff
@@ -0,0 +1,241 @@
+/*
+ * Copyright (c) 2018, Franck Maillot <https://github.com/Franck-M>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.kourendlibrary;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import net.runelite.api.coords.WorldPoint;
+import static net.runelite.client.plugins.kourendlibrary.Book.KILLING_OF_A_KING;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.runners.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class KourendLibraryTest
+{
+	private Library library = new Library();
+
+	private Map<Bookcase, Book> bookByBookcase;
+
+	private Book activeDarkManuscript;
+
+	private void fillBookcases(List<Book> sequence, int startingIndex, Book activeDarkManuscript)
+	{
+		this.activeDarkManuscript = activeDarkManuscript;
+		bookByBookcase = new HashMap<>();
+		int step = library.getStep();
+		List<Bookcase> byIndex = library.getBookcaseByIndex();
+		for (int i = 0; i < byIndex.size(); i++)
+		{
+			int ai = (i + startingIndex) % byIndex.size();
+			Book seqBook = ((i % step == 0) && (i / step < sequence.size())) ? sequence.get(i / step) : null;
+			Bookcase iBookcase = byIndex.get(ai);
+
+			if (seqBook != null)
+			{
+				assert !bookByBookcase.containsKey(iBookcase);
```

Comment:
It is preferred to use Junit's `Assert` class incase the tests are ran with assertions disabled